### PR TITLE
[WIN32SS][NTGDI][GDI32] Fix DPtoLP and NtGdiTransformPoints etc.

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Fix `NtGdiTransformPoints`, `gdi32!DPtoLP`, `gdi32!LPtoDP` and `gdi32!SetMapMode` functions. Fix CORE-15983.
JIRA issue: [CORE-15983](https://jira.reactos.org/browse/CORE-15983)

### CODE REVIEW

**Comment by katahiromz** [comment]
BEFORE:
DPtoLP-before
AFTER:
2nd-trial

File: win32ss/gdi/ntgdi/coord.h [hunk 1]
```diff
@@ -131,6 +131,9 @@ DC_vXformDeviceToWorld(
     PMATRIX pmx;
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/coord.h]: Note: pptlDest is OUT.

File: win32ss/gdi/ntgdi/coord.h [hunk 2]
```diff
@@ -131,6 +131,9 @@ DC_vXformDeviceToWorld(
...
 
     pmx = DC_pmxDeviceToWorld(pdc);
+    if (!MX_IsValid(pmx))
+        return;
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/coord.h]: What does pptlDest contain if this check fails? (-> invalid value?)

File: win32ss/gdi/ntgdi/xformobj.c [hunk 3]
```diff
@@ -243,6 +244,25 @@ XFORMOBJ_iCombineXform(
     }
 }
 
+BOOL FASTCALL
+MX_IsValid(IN PMATRIX pmx)
+{
+    FLOATOBJ foDet;
+    MulSub(&foDet, &pmx->efM11, &pmx->efM22, &pmx->efM12, &pmx->efM21);
```
Comment (HBelusca) [inline win32ss/gdi/ntgdi/xformobj.c]: What are you checking here? If you check whether the determinant of the matrix is not equal to zero, then you could rename your function into "MX_IsInvertible".

Comment (katahiromz) [reply win32ss/gdi/ntgdi/xformobj.c]: OK. Renamed.

File: win32ss/gdi/ntgdi/coord.h (hunk 2 above)
Comment (katahiromz) [reply win32ss/gdi/ntgdi/coord.h]: I have no idea.

Comment (katahiromz) [reply win32ss/gdi/ntgdi/coord.h]: Maybe it needs a testcase of `NtGdiTransformPoints`.

Comment (katahiromz) [reply win32ss/gdi/ntgdi/coord.h]: I added a testcase of `NtGdiTransformPoints` function in #1542.

**Comment by katahiromz** [comment]
transform-fixed

**Comment by katahiromz** [comment]
Big fix.
fixed

**Comment by katahiromz** [comment]
map-mode

**Comment by katahiromz** [comment]
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24461

**Comment by katahiromz** [comment]
https://build.reactos.org/builders/Build%20GCCLin_x86/builds/24463

**Comment by BenNottelling** [comment]
Can I just say I appreciate that you provide a link to the JIRA issue as it makes it easier to check by and see what issue is being fixed quickly and easily (as I'm not really a dev)

**Pull request merged**